src/utils/useOutsideClickListener.js

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+
+function useOutsideClickListener(ref, isActive, onToggle) {
+  function handleOutsideClick(event) {
+    const target = event.target || {};
+    if (!ref.current.contains(target)) {
+      onToggle();
+      window.removeEventListener('click', handleOutsideClick, {
+        capture: true,
+      });
+    }
+  }
+
+  useEffect(() => {
+    if (isActive) {
+      window.addEventListener('click', handleOutsideClick, { capture: true });
+    } else {
+      window.removeEventListener('click', handleOutsideClick, {
+        capture: true,
+      });
```

Comment:
this seems to be duplicated. I am guessing we can have a single function that abstracts this then can be consumed in the respective places.